src/WikiService.Validation/Rubric/EditRubricRequestValidator.cs

```diff
@@ -0,0 +1,195 @@
+      RequestedOperation = requestedOperation;
+
+      #region paths
+
+      AddСorrectPaths(
+        new List<string>
+        {
+          nameof(EditRubricRequest.Name),
+          nameof(EditRubricRequest.ParentId),
+          nameof(EditRubricRequest.IsActive)
+        });
+
+      AddСorrectOperations(nameof(EditRubricRequest.Name), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditRubricRequest.ParentId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditRubricRequest.IsActive), new List<OperationType> { OperationType.Replace });
+
+      #endregion
+
+      #region Name
+
+      AddFailureForPropertyIf(
+        nameof(EditRubricRequest.Name),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            x => !string.IsNullOrWhiteSpace(x.value?.ToString()),
+            "Name must not be empty."
+          },
+          {
+            x => x.value.ToString().Trim().Length <= 150,
+            "Name is too long."
+          }
+        }, CascadeMode.Stop);
+
+      #endregion
+
+      #region IsActive
+
+      AddFailureForPropertyIf(
+       nameof(EditRubricRequest.IsActive),
+       x => x == OperationType.Replace,
+       new()
+       {
+         {
+           x => bool.TryParse(x.value?.ToString(), out bool _), "Incorrect rubric is active format"
+         },
+       }, CascadeMode.Stop);
+
+      #endregion
+
+      #region ParentId
+
+      await AddFailureForPropertyIfAsync(
+        nameof(EditRubricRequest.ParentId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            x => Task.FromResult(
+              string.IsNullOrEmpty(x.value?.ToString())
+              ? true
+              : Guid.TryParse(x.value.ToString(), out Guid result)),
+            "Incorrect format of ParentId."
+          },
+          {
+            async (x) =>
+            {
+              if (x.value?.ToString() is null)
+              {
+                return true;
+              }
+
+              if (!Guid.TryParse(x.value?.ToString(), out Guid parentId))
+              {
+                return false;
+              }
+
+              return await _rubricRepository.DoesExistAsync(parentId);
+            },
+            "Parent id doesn`t exist."
+          }
+        }, CascadeMode.Stop);
+    }
+
+    #endregion
+
+    public EditRubricRequestValidator(
+        IRubricRepository rubricRepository)
+    {
+      _rubricRepository = rubricRepository;
+
+      RuleForEach(x => x.Item2.Operations)
+        .CustomAsync(async (x, context, _) => await HandleInternalPropertyValidationAsync(x, context));
+
+      When(x => x.Item2.Operations.Any(o => o.path.EndsWith(nameof(EditRubricRequest.ParentId), StringComparison.OrdinalIgnoreCase)) == true
+        && x.Item2.Operations.Any(o => o.path.EndsWith(nameof(EditRubricRequest.Name), StringComparison.OrdinalIgnoreCase)) == true,
```

Comment:
you need this check not only when Name AND RubricId are changing. but when Name OR RubricId are changing too.
So, not "&&", but "||"